core/types/receipt.go

```diff
@@ -610,16 +610,24 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 		rs[i].Type = txs[i].Type()
 		rs[i].TxHash = txs[i].Hash()
 
-		// Pre-gingerbred the base fee was stored in state, but we don't try to recover it here, since A) we don't have
-		// access to the objects required to get the state and B) retrieving the base fee is quite code heavy and we
-		// don't want to bring that code across from the celo L1 to op-geth. In the celo L1 we would return a nil base
-		// fee if the state was not available, so that is what we do here.
-		if config.IsGingerbread(new(big.Int).SetUint64(number)) {
-			// The post transition CeloDynamicFeeV2Txs set the baseFee in the receipt
-			if rs[i].BaseFee == nil {
-				rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)
-			} else {
-				rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), rs[i].BaseFee)
+		switch rs[i].Type {
+		case LegacyTxType, AccessListTxType:
+			// These are the non dynamic tx types so we can simply set effective gas price to gas price.
+			rs[i].EffectiveGasPrice = new(big.Int).Set(txs[i].GasPrice())
```

Comment:
Those transaction already implement that behaviour: https://github.com/celo-org/op-geth/blob/c4a688130ae3b57dde330428c51c47432df54ce0/core/types/tx_legacy.go#L126-L128

I'd rather use `txs[i].inner.effectiveGasPrice(new(big.Int), basefee)` here.

Reply:
Why this preference?

Reply:
For me it feels better to use the protocol instead of "reimplementing" it.
Upstream and the default branch use `effectiveGasPrice`, so I'd use it here as well if possible.

Reply:
Ok, yeah I was thinking that it's a bit obfuscated to pass the base fee and then ignore it. But happy to change to that approach.